File: Programmers/Level2/행렬테두리회전하기/행렬테두리회전하기.py
def solution(rows, columns, queries):
    answer = []
    matrix = [[0 for _ in range(columns + 1)]for _ in range(rows + 1)]
    
    value = 1
    for r in range(1, rows + 1):
        for c in range(1, columns + 1):
            matrix[r][c] = value
            value += 1
            
    for x1, y1, x2, y2 in queries:
        tmp_value = matrix[x1][y1]
        min_value = tmp_value
        
        for i in range(x1, x2):
            tmp = matrix[i + 1][y1]
            matrix[i][y1] = tmp
            min_value = min(min_value, tmp)
        
        for i in range(y1, y2):
            tmp = matrix[x2][i + 1]
            matrix[x2][i] = tmp
            min_value = min(min_value, tmp)
            
        for i in range(x2, x1, -1):
            tmp = matrix[i - 1][y2]
            matrix[i][y2] = tmp
            min_value = min(min_value, tmp)
            
        for i in range(y2, y1, -1):
            tmp = matrix[x1][i - 1]
            matrix[x1][i] = tmp
            min_value = min(min_value, tmp)
        
        matrix[x1][y1 + 1] = tmp_value
        answer.append(min_value)
        
    return answer

File: Programmers/Level2/행렬테두리회전하기/행렬테두리회전하기.js
function solution(rows, columns, queries) {
    answer = [];
    matrix = [...Array(rows + 1)].map(() => [...Array(columns + 1)].fill(0));
    let v = 1;
    for (let r = 1; r < rows + 1; r++) {
        for (let c = 1; c < columns + 1; c++) {
            matrix[r][c] = v;
            v += 1;
        }
    }

    for (let q of queries) {
        let [x1, y1, x2, y2] = q;
        let check = matrix[x1][y1];
        let _min = check;

        // 왼쪽 세로
        for (let i = x1; i < x2; i++) {
            let tmp = matrix[i + 1][y1];
            matrix[i][y1] = tmp;
            _min = Math.min(_min, tmp);
        }

        // 하단 가로
        for (let i = y1; i < y2; i++) {
            let tmp = matrix[x2][i + 1];
            matrix[x2][i] = tmp;
            _min = Math.min(_min, tmp);
        }

        // 오른쪽 세로
        for (let i = x2; i > x1; i--) {
            let tmp = matrix[i - 1][y2];
            matrix[i][y2] = tmp;
            _min = Math.min(_min, tmp);
        }

        // 상단 가로
        for (let i = y2; i > y1; i--) {
            let tmp = matrix[x1][i - 1];
            matrix[x1][i] = tmp;
            _min = Math.min(_min, tmp);
        }
        matrix[x1][y1 + 1] = check;
        answer.push(_min);
    }
    return answer;
}
